launcher/main/main.go

```diff
@@ -106,8 +106,17 @@ func main() {
 		fmt.Println("Failed to assemble executable metadata", cmds, err)
 		panic(err)
 	}
+	var primaryCmd *exec.Cmd
+	subProcesses := make(map[string]*exec.Cmd)
+	for _, cmd := range cmds {
+		if cmd.Name == "primary" {
```

Comment:
I agree this is smelly, and I knew that when I wrote it -  here was my thought process behind this decision: `go-java-launcher` needs to know which command should be run as the primary process, but `go-init` doesn't (since each process is bundled with its output filename). The return value from `CompileCmdsFromConfigFiles` (the struct) will conform to one of those use cases better than it will the other, and the other use case will have to do something suboptimal to account for that. (`go-java-launcher` has to do what is written here; `go-init` has to do something that is, admittedly, substantially less suboptimal, but is still a bit weird - it has to apply the same logic to an array of a type as well as just a single value of that type, whereas it would be cleaner if it just received an array that also included that single value).

Given the choice of which use case I wanted to tailor this to, I chose `go-init`, since that's our long-term solution, and `go-java-launcher` launching multiple processes is a short-term solution. One possible way to get around this that allows us to have `go-init` still be really clean and eliminate this smelliness is to just remove the `go-java-launcher` multiprocess stuff as part of this PR?